### PR TITLE
filezilla: more portable

### PR DESCRIPTION
**Pull request by wzv5**

### CODE REVIEW

File: bucket/filezilla.json [hunk 1]
```diff
@@ -15,19 +15,31 @@
...
-        "    Copy-Item \"$dir\\docs\\fzdefaults.xml.example\" \"$dir\\fzdefaults.xml\"",
+        "if (-not (Test-Path \"$persist_dir\\fzdefaults.xml\")) {",
+        "$finalFile = '<FileZilla3><Settings>'",
+        "$finalFile += '<Setting name=\"Config Location\"></Setting>'",
+        "$finalFile += '<Setting name=\"Cache directory\"></Setting>'",
+        "$finalFile += '</Settings></FileZilla3>'",
+        "Set-Content \"$dir\\fzdefaults.xml\" -Value $finalFile -Encoding Ascii",
         "}"
     ],
+    "post_install": [
```
Comment (Ash258) [inline bucket/filezilla.json]: Why post install?? 
Just set it ini L20, L21.

File: bucket/filezilla.json [hunk 2]
```diff
@@ -15,19 +15,31 @@
     },
     "extract_dir": "FileZilla-3.42.1",
     "pre_install": [
-        "$file = 'fzdefaults.xml'",
-        "if (-not (Test-Path \"$persist_dir\\$file\")) {",
-        "    Copy-Item \"$dir\\docs\\fzdefaults.xml.example\" \"$dir\\fzdefaults.xml\"",
+        "if (-not (Test-Path \"$persist_dir\\fzdefaults.xml\")) {",
+        "$finalFile = '<FileZilla3><Settings>'",
+        "$finalFile += '<Setting name=\"Config Location\"></Setting>'",
+        "$finalFile += '<Setting name=\"Cache directory\"></Setting>'",
```
Comment (Ash258) [inline bucket/filezilla.json]:
```suggestion
        "$finalFile += \"<Setting name=\"\"Cache directory\"\">$dir\\data\\cache</Setting>\"",
```

File: bucket/filezilla.json [hunk 3]
```diff
@@ -15,19 +15,31 @@
     },
     "extract_dir": "FileZilla-3.42.1",
     "pre_install": [
-        "$file = 'fzdefaults.xml'",
-        "if (-not (Test-Path \"$persist_dir\\$file\")) {",
-        "    Copy-Item \"$dir\\docs\\fzdefaults.xml.example\" \"$dir\\fzdefaults.xml\"",
+        "if (-not (Test-Path \"$persist_dir\\fzdefaults.xml\")) {",
+        "$finalFile = '<FileZilla3><Settings>'",
+        "$finalFile += '<Setting name=\"Config Location\"></Setting>'",
```
Comment (Ash258) [inline bucket/filezilla.json]:
```suggestion
        "$finalFile += \"<Setting name=\"\"Config Location\"\">$dir\\data\\config</Setting>\"",
```

File: bucket/filezilla.json (hunk 1 above)
Comment (wzv5) [reply bucket/filezilla.json]: `pre_install` is for the first install, just create a default config file.
`post_install` is used to keep the location forever.

Comment (Ash258) [reply bucket/filezilla.json]: Completely useless.

Do it in pre_install block as suggestions.
Becuase in post_install you edit this file all the time. So when someone do not want his config and cache directory inside install path and change it (in GUI or config file) you will completely break his installation

Comment (wzv5) [reply bucket/filezilla.json]: You are right.
But when I changed to `pre_install`, I found that `$dir` is contained version numbers, not `current`.
I can't solve this problem. Please help me.

Comment (Ash258) [reply bucket/filezilla.json]: various solutions:

1. Split-Path $dir | Join-Path -ChildPath 'current'
1. $(appdir <manifest> $global)/current'
1. current_dir $dir

**Comment by wzv5** [comment]
@Ash258 I have made a forced push, please have a look.

**Comment by wzv5** [comment]
Closed due to no response.

**Comment by rasa** [comment]
These changes appear worth considering, but I can't reopen.

**Pull request closed without merging**